lib/ClangImporter/ClangImporter.cpp

```diff
@@ -4206,9 +4207,22 @@ ClangImporter::getSwiftExplicitModuleDirectCC1Args() const {
 
   if (!Impl.SwiftContext.SearchPathOpts.ScannerPrefixMapper.empty()) {
     // Remap all the paths if requested.
+    // FIXME: Should we also do
+    // https://github.com/swiftlang/llvm-project/pull/10723?
     llvm::PrefixMapper Mapper;
-    clang::tooling::dependencies::DepscanPrefixMapping::configurePrefixMapper(
-        Impl.SwiftContext.SearchPathOpts.ScannerPrefixMapper, Mapper);
+    SmallVector<llvm::MappedPrefix> Prefixes;
+    if (auto E = llvm::MappedPrefix::transformJoined(
+            Impl.SwiftContext.SearchPathOpts.ScannerPrefixMapper, Prefixes)) {
+      // Save this string. In general the diagnostic might outlive this
+      // function.
+      auto identifier =
+          Impl.SwiftContext.getIdentifier(llvm::toString(std::move(E)));
+      Impl.SwiftContext.Diags.diagnose(SourceLoc(), diag::error_prefix_mapping,
+                                       identifier.str());
+    }
+    Mapper.addRange(Prefixes);
+    Mapper.sort();
+
```

Comment:
This is probably the only change that I feel needs a review. @Xazax-hun Could you take a look?


See https://github.com/swiftlang/llvm-project/pull/10723.

Reply:
Took a look, apart from a nit looks good to me. But just to double check, is this helper function that we used to use gone without any replacement? Although now at least we have a diagnostic for the failure case, which is great.